compress.cc

```diff
@@ -113,6 +108,22 @@ compression_parameters::compression_parameters(compressor_ptr c)
     : _compressor(std::move(c))
 {}
 
+std::string_view compression_parameters::algorithm_to_name(algorithm alg) {
+    switch (alg) {
+        case algorithm::lz4: return "LZ4Compressor";
+        case algorithm::deflate: return "DeflateCompressor";
+        case algorithm::snappy: return "SnappyCompressor";
+        case algorithm::zstd: return "ZstdCompressor";
+        case algorithm::none: abort();
+    }
```

Comment:
And on_internal_error here too.

Reply:
You mean, add `on_internal_error()` after the switch? I'll add `abort()` there instead.

The compiler will complain if there's a path where a function doesn't return. Which means that we can't go past the switch due to a "benign" programmer error. It can only happen if the enum has an illegal value. And that can only be caused by memory corruption. In this situation, `abort()` makes more sense than `on_internal_error()`.

Reply:
Added `abort()` there.